src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs

```diff
@@ -610,55 +936,25 @@ public SocketError Accept(byte[] socketAddress, ref int socketAddressLen, int ti
             Debug.Assert(timeout == -1 || timeout > 0, $"Unexpected timeout: {timeout}");
 
             SocketError errorCode;
-            if (SocketPal.TryCompleteAccept(_socket, socketAddress, ref socketAddressLen, out acceptedFd, out errorCode))
+            int observedSequenceNumber;
+            if (_receiveQueue.IsReady(this, out observedSequenceNumber) &&
+                SocketPal.TryCompleteAccept(_socket, socketAddress, ref socketAddressLen, out acceptedFd, out errorCode))
```

Comment:
BTW, it does concern me that we take a lock in IsReady, since this happens on every IO op.  I considered trying to combine the queue state and seq number into a single field that we could update using interlocked ops -- then, we could just do a volatile read in IsReady, since it doesn't examine the queue at all.  However, other callers still need to lock to protect the queue, and that means they would need to lock *and* do interlocked on the state/seq#, which doesn't seem great.

I'd love to improve this, just not sure how right now.  Thoughts?